### PR TITLE
Fix parsing of generatedfilesout option

### PR DESCRIPTION
**Pull request by chsienki**
Fixes https://github.com/dotnet/roslyn/issues/49981

We were failing to remove extra quotes from the `generatedfilesout` argument. It worked fine when passed directly via the commandline as the OS removed them for us, but failed in response file scenarios such as when compiling via the build server.

### CODE REVIEW

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -12747,6 +12747,83 @@ class C
             Directory.Delete(dir.Path, true);
         }
 
+        [Fact]
+        public void SourceGenerators_GeneratedDir_Has_Spaces()
```
Comment (chsienki) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:12751]: This test would actually pass without the fix, but I figured it was worth adding the extra coverage anyway.

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 2]
```diff
@@ -12747,6 +12747,82 @@ class C
...
```
Comment (chsienki) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:12751]: This test would actually pass without the fix, but I figured it was worth adding the extra coverage anyway.

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for a small arg parsing fix.

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 3]
```diff
@@ -12747,6 +12747,82 @@ class C
...
+            var generatorPrefix = GeneratorDriver.GetFilePathPrefixForGenerator(generator);
+            ValidateWrittenSources(new() { { Path.Combine(generatedDir.Path, generatorPrefix), new() { { "generatedSource.cs", generatedSource } } } });
+
+            // Clean up temp files
+            CleanupAllGeneratedFiles(src.Path);
+            Directory.Delete(dir.Path, true);
+        }
+
+        [Fact]
+        public void ParseCompilerGeneratedFilesOut()
```
Comment (cston) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:12774]: >CompilerGeneratedFilesOut [](start = 25, length = 25)

"GeneratedFilesOut"?

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 4]
```diff
@@ -12747,6 +12747,82 @@ class C
...
+        {
+            string root = PathUtilities.IsUnixLikePlatform ? "/" : "c:\\";
+            string baseDirectory = Path.Combine(root, "abc", "def");
+
+            var parsedArgs = DefaultParse(new[] { @"/generatedfilesout:", "a.cs" }, baseDirectory);
+            parsedArgs.Errors.Verify(
+                // error CS2006: Command-line syntax error: Missing '<text>' for '/compilergeneratedfilesout:' option
```
Comment (cston) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:12781]: >/compilergeneratedfilesout: [](start = 82, length = 27)

`/generatedfilesout:` here and below.

**Review by cston — APPROVED** [review]

**Pull request merged**